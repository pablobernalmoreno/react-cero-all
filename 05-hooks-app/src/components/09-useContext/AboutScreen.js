import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser({})}>
        Clear
      </button>
    </div>
  );
};
