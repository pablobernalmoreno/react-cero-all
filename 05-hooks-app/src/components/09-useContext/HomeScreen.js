import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>HomeScreen </h1>
      <h2>{JSON.stringify(user)}</h2>
      <hr />
    </div>
  );
};
