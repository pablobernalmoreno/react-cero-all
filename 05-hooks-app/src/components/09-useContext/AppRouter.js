import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Route path="/" component={HomeScreen} />
          <Route component={HomeScreen} />
        </Switch>
      </div>
    </Router>
  );
};
