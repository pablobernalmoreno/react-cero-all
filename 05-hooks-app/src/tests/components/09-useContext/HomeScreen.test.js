import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en HomeScreen", () => {
  const user = {
    name: "Pablo",
    email: "pablioto@gmail.com",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user: user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
