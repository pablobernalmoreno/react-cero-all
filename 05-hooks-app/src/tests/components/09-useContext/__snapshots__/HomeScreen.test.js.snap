// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen debe de mostrarse correctamente 1`] = `
<HomeScreen>
  <div>
    <h1>
      HomeScreen 
    </h1>
    <h2>
      {"name":"Pablo","email":"pablioto@gmail.com"}
    </h2>
    <hr />
  </div>
</HomeScreen>
`;
